packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -882,18 +907,25 @@ function updateDynamicChildren(
         } else if (isSameVnode(oldStartVnode, newEndVnode)) {
             // Vnode moved right
             patch(oldStartVnode, newEndVnode, parent, renderer);
-            insertNode(
-                oldStartVnode.elm!,
-                parent,
-                renderer.nextSibling(oldEndVnode.elm!),
-                renderer
-            );
+
+            // In the case of fragments, the `elm` property of a vfragment points to the leading
+            // anchor. To determine the next sibling of the whole fragment, we need to use the
+            // trailing anchor as the argument to nextSibling():
+            // [..., [leading, ...content, trailing], nextSibling, ...]
+            let anchor: Node | null;
+            if (isVFragment(oldEndVnode)) {
```

Comment:
Minor nit: we could save a check here if `trailing` was on VNode rather than Vfragment because we can do
```javascript
anchor = renderer.nextSibling(oldEndVnode.trailing ? oldEndVnode.trailing.elm : oldEndVnode.elm)
```
but I don't think we need to hold up this PR for that change. We can consider that when refactoring.